crates/core/src/delta_datafusion/mod.rs

```diff
@@ -765,8 +766,26 @@ impl<'a> DeltaScanBuilder<'a> {
             .parquet_options
             .unwrap_or_else(|| self.session.table_options().parquet.clone());
 
+        // We have to set the encryption factory on the ParquetSource based on the Parquet options,
+        // as this is usually handled by the ParquetFormat type in DataFusion,
+        // which is not used in delta-rs.
```

Comment:
This feels a bit convoluted, we could possible set an `EncryptionFactory` directly in the format options rather than going through the DataFusion config if we just have to set the encryption factory ourselves anyway. But it's nice that at the moment `FileFormatOptions` doesn't have anything specific to encryption in it. I guess we could add a method to `FileFormatOptions` to modify a `ParquetSource` instead, or create one.